### PR TITLE
refactor(button): Refactor button to use mdl-theme.

### PR DESCRIPTION
**Pull request by sgomes**
@Garbee PTAL!

### CODE REVIEW

**Comment by Garbee** [comment]
Commit is authorized and CLA is already signed.

**Comment by Garbee** [comment]
@traviskaufman Can you please verify this merge?

Looks good to me as-is (since I just pushed the needed changed for dark mode shading.)

File: packages/mdl-button/mdl-button.scss [hunk 1]
```diff
@@ -96,15 +105,34 @@
...
+      @include mdl-theme-prop(background-color, primary);
+
+      /* postcss-bem-linter: ignore */
+      &::before {
+        /*
+          We are explicitly not fully adhering to Material Design here.
+          This should be the 700-shade when active instead of a black shade.
+          Due to the complexity involved in adhering fully it is being ignored.
+          Instead re-using the existing architecture for shading works just fine.
+        */
```
Comment (traviskaufman) [inline packages/mdl-button/mdl-button.scss]: 👍

**Comment by traviskaufman** [comment]
💎 LGTM

**Pull request merged**